flink-runtime/src/main/java/org/apache/flink/runtime/source/coordinator/SourceCoordinatorContext.java

```diff
@@ -220,8 +223,20 @@ void sendEventToSourceOperator(int subtaskId, OperatorEvent event) {
                 String.format("Failed to send event %s to subtask %d", event, subtaskId));
     }
 
-    ScheduledExecutorService getCoordinatorExecutor() {
-        return coordinatorExecutor;
+    void sendEventToSourceOperatorIfTaskReady(int subtaskId, OperatorEvent event) {
+        checkSubtaskIndex(subtaskId);
+
+        callInCoordinatorThread(
```

Comment:
What happens in `callInCoordinatorThread` if there is an unhandled exception in the `callable`? Does this failover the job?

Reply:
No, it just throw the exception that thrown by callable~ @pnowojski